pysat/tests/classes/cls_instrument_access.py

```diff
@@ -0,0 +1,833 @@
+        """Ensure doy load argument in valid range."""
+
+        with pytest.raises(ValueError) as err:
+            self.testInst.load(self.ref_time.year, doy)
+        estr = 'Day of year (doy) is only valid between and '
+        assert str(err).find(estr) >= 0
+
+        return
+
+    @pytest.mark.parametrize('end_doy', [0, 367, 1000, -1, -10000])
+    def test_basic_instrument_load_yr_bad_end_doy(self, end_doy):
+        """Ensure end_doy keyword in valid range."""
+
+        with pytest.raises(ValueError) as err:
+            self.testInst.load(self.ref_time.year, 1, end_yr=self.ref_time.year,
+                               end_doy=end_doy)
+        estr = 'Day of year (end_doy) is only valid between and '
+        assert str(err).find(estr) >= 0
+
+        return
+
+    def test_basic_instrument_load_yr_no_end_doy(self):
+        """Ensure end_doy required if end_yr present."""
+
+        with pytest.raises(ValueError) as err:
+            self.testInst.load(self.ref_time.year, self.ref_doy,
+                               self.ref_time.year)
+        estr = 'Both end_yr and end_doy must be set'
+        assert str(err).find(estr) >= 0
+
+        return
+
+    @pytest.mark.parametrize("input", [{'yr': 2009, 'doy': 1,
+                                        'date': dt.datetime(2009, 1, 1)},
+                                       {'yr': 2009, 'doy': 1,
+                                        'end_date': dt.datetime(2009, 1, 1)},
+                                       {'yr': 2009, 'doy': 1,
+                                        'fname': 'dummy_str.nofile'},
+                                       {'yr': 2009, 'doy': 1,
+                                        'stop_fname': 'dummy_str.nofile'},
+                                       {'date': dt.datetime(2009, 1, 1),
+                                        'fname': 'dummy_str.nofile'},
+                                       {'date': dt.datetime(2009, 1, 1),
+                                        'stop_fname': 'dummy_str.nofile'},
+                                       {'date': dt.datetime(2009, 1, 1),
+                                        'fname': 'dummy_str.nofile',
+                                        'end_yr': 2009, 'end_doy': 1}])
+    def test_basic_instrument_load_mixed_inputs(self, input):
+        """Ensure mixed load inputs raise ValueError."""
+
+        with pytest.raises(ValueError) as err:
+            self.testInst.load(**input)
+        estr = 'An inconsistent set of inputs have been'
+        assert str(err).find(estr) >= 0
+        return
+
+    def test_basic_instrument_load_no_input(self):
+        """Test that `.load()` loads all data."""
+
+        self.testInst.load()
+        assert (self.testInst.index[0] == self.testInst.files.start_date)
+        assert (self.testInst.index[-1] >= self.testInst.files.stop_date)
+        assert (self.testInst.index[-1] <= self.testInst.files.stop_date
+                + dt.timedelta(days=1))
```

Comment:
Is this the same as the eval method?

Reply:
No, because this checks against the whole range set by bounds